refactor: use click for cli instead of argv parsing

rehmanis opened the pull request:
Fixes #473, #474, #475

Instead of using argv parsing we now use `click`. 

- [x] add a new file `cli.py` that will be used for all click commands (currently in `cmd.py`)
- [x] refactor `shopyo clean` to `shopyo clean [OPTIONS]` with no argv parsing 
- [x] write tests for `shopyo clean [OPTIONS]`.
- [x] refactor `shopyo startbox BOXNAME` to `shopyo startbox [OPTIONS] BOXNAME` with no argv parsing 
- [x] write tests for `shopyo startbox [OPTIONS] BOXNAME`.
- [x] refactor `shopyo initialise` to `shopyo initialise [OPTIONS]`.
- [x] refactor `shopyo collecstatic` to `shopyo collectstatic [OPTIONS] [SRC]`.
- [x] write tests for `shopyo collectstatic [OPTIONS] [SRC]`.
- [x] add option to pass app config type for any `shopyo` command
- [x] refactor `shopyo startapp MODULENAME`, `shopyo startbox BOXNAME` and `shopyo startsubapp MODULENAME in BOXNAME` to  one command `shopyo createmodule [OPTIONS] MODULENAME [BOXNAME]` 
- [x] write tests for `shopyo createmodule [OPTIONS] MODULENAME [BOXNAME]` 
- [x] refactor `shopyo new PROJNAME` to `shopyo new [OPTIONS] [PROJNAME]`.
- [x] commands can be called via `python manage.py cmd` 
- [x] allow users to put their own cli command. Example, if user runs `shopyo new blog` and then does `cd blog/blog`, and runs `python -m pip install -e .`, he can then run commands like `blog --help` or `blog --version` and then add more commands in `cli.py`. A sample command is already added and can be run as follows `blog welcome [OPTIONS] NAME`
- [x] write tests for `shopyo new [OPTIONS] [PROJNAME]`.
- [x] update docs

Run these in `shopyo/shopyo` to see all the available commands
```bash
$ shopyo
```
or 
```bash
$ shopyo --help
```
Then run for example `$ shopyo <cmd> --help` to see the `<cmd>` usage. The refactored commands can also be run as `python manage.py clean` for example

Update:
New command now has an option to not enter the project name in which case the project of same name as parent folder will be created in current working directory.

rehmanis committed:
refactor: use click for cli insteade of argv parsing

Instead of using argv parsing we now use click. For now only clean2 and startbox2 commmands are written to replace clean and startbox command. Added same tests as before for clean cmd but now we invovke the clean cmd in the tests. updated the setup.py entrypoint to now point to the new file script.py containing all click cmds. New command will not work for now as entrypoint has been changed

pep8speaks commented:
Hello @rehmanis! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







* In the file [`shopyo/api/cli_content.py`](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py):

> [Line 91:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L91): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 92:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L92): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 180:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L180): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 186:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L186): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 198:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L198): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 199:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L199): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 200:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L200): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 208:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L208): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 211:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L211): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 212:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L212): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 217:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L217): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 347:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L347): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 504:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L504): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 508:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L508): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 510:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L510): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 511:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L511): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 512:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L512): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 522:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L522): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 529:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L529): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 541:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L541): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 550:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L550): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 562:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L562): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 565:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L565): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 574:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L574): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 575:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L575): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 576:80](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/cli_content.py#L576): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)





* In the file [`shopyo/api/database.py`](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/database.py):

> [Line 49:19](https://github.com/Abdur-rahmaanJ/shopyo/blob/97371d24f2fb657839e730fae221cf01414537a8/shopyo/api/database.py#L49): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file



























##### Comment last updated at 2021-04-01 05:20:41 UTC

rehmanis committed:
test: add test for startbox2 cmd

rehmanis committed:
refactor: clean and write the shopyo initialise cmd

-moved the file helpers to file.py

-updated the printouts for refactored initialise

-added more private cmd wrappers for collectstatic and uploads

-added a constant.py file for contants used for printing output from cmd

rehmanis committed:
merge: get the latest changes and save the current work on cli refactoring

rehmanis committed:
resolve merge conflicts

rehmanis committed:
refactor: add the new click command for collectstatic to replace the old on

- added tests for the new collectstatic command

rehmanis committed:
fix the collecstatic to work for both Unix vs mindows path

rehmanis committed:
refactor: add createmodule command

- the new commands replaces startapp and startsubapp

the new command creates the boxname and module using one command

- change the test for startbox2 cmd. will remove this cmd later since createmodule cmd covers the startbox2 cmd as well

rehmanis committed:
tests: add createmodule cli tests

rehmanis committed:
refactor: new2 cli now added with basic tests

-new2 command now works but is still incomplete as its missing some files added in old new command.

-tests are still incomplete for new2

rehmanis committed:
refactor: update manage.py to match new cli

-cmd.py code and tests as well as old manage.py code is commented for now. Will be removed before merge

rehmanis committed:
refactor: new command comaplete.

bugs: fixed bugs in the old command such as pytest failing

feat: cli.py is added
that allows users to add their own project command

tox still fails since it is using old shopyo release

-commented the __main__.py
for now. Will remove the commented out code before merge.

rehmanis committed:
refactor: remove commented out code

rehmanis committed:
refactor: delete unused files

rehmanis committed:
refactor: rename files and commmands

-script.py is rename to cli.py, and  in the commands suffix has been removed

rehmanis committed:
tests: new command unit tests are completed

-updated new command to now have an option to not put the projectname in which case it will create the project of same name as the parent folder in current directory

rehmanis committed:
docs: add cli api for shopyo

rehmanis committed:
docs: update commandline sections

rehmanis committed:
docs: remove unused extentions from sphinx conf.py

rehmanis committed:
docs: update all old commands with new

-modified the Modules/App, and assets section

rehmanis committed:
docs: fix usage information in commandline

rehmanis committed:
refactor: add the shopyo info message before each command

rehmanis committed:
config now changes the FLASK_ENV and FLASK_APP variable. Docs update complete

rehmanis commented:
@Abdur-rahmaanJ All done. Ready for review.  I used the wrong branch name for this. Should I change it somehow or is it okay this way?

rehmanis committed:
fix: adding FLASK_APP is causing importError which causes shopyo initialise to fail. Commented that part for now
